### ISSUE TITLE
Overridding attributes in auto-structs results in duplicated attributes

### ISSUE THREAD

**Issue by solnic** [rom-rb/rom#491]
``` ruby
module Entities
  class User < ROM::Struct
    attribute :login, Types::String
  end
end

class UserRepo < ROM::Repository[:users]
  struct_namespace Entities

  def by_id(id)
    users.by_pk(id).one
  end
end

user = rom.relations[:users]
  .changeset(:create, login: 'jane@doe.org', email: 'jane@doe.org')
  .commit

repo = UserRepo.new(rom)

repo.by_id(1)
# => #<Entities::User login="jane@doe.org" id=1 login="jane@doe.org" email="jane@doe.org">
```

Notice duplicated `login` in the inspected user. This is either a bug in the way we set up `Equalizer` in dry-struct OR we actually end up with the same attribute defined multiple times (which would be way worse).

**Comment by flash-gordon** [comment]
This is a bug in dry-struct. https://github.com/dry-rb/dry-struct/blob/v0.4.0/lib/dry/struct/class_interface.rb#L73
It is fixed in master but rom isn't compatible with it. We could cut a branch from `v0.4.0` if needed

**Comment by solnic** [comment]
ahh OK, could we push `0.4.1` with just this fix (or any other changes that are safe to have in this release)?

**Comment by flash-gordon** [comment]
we could, this will be a separate fix though since we don't have this problem in master as we don't have that code at all :)

**Comment by solnic** [comment]
OK, so I guess this can wait for the 5.0 upgrade, unless somebody would like to fix it in `0.4.x`
